tests/end2end/features/tabs.feature

```diff
@@ -1026,3 +1026,100 @@ Feature: Tab management
             - tabs:
               - history:
                 - url: http://localhost:*/data/hello.txt
+
+    # :tab-pin
+
+    Scenario: :tab-pin command
+        When I open data/numbers/1.txt
+        And I open data/numbers/2.txt in a new tab
+        And I open data/numbers/3.txt in a new tab
+        And I run :tab-pin
```

Comment:
There's Python code running behind each of those "When ..." and "Then ..." lines. qutebrowser is run in a subprocess, and the tests send commands to it and inspect files or the log output. In case of "Then the following tabs should be open:" (`tests/end2end/features/conftest.py:check_open_tabs`) it gets a session file from qutebrowser and checks that against the given list.

I'd just add something similar to `(active)` there, then you can easily check it via that command.